onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -2630,6 +2637,40 @@ def aten_div(self: TFloat, other: TFloat) -> TFloat:
     return op.Div(self, other)
 
 
+@torch_op(
+    (
+        "aten::div",
+        "aten::div.Tensor",
+        "aten::div.Scalar",
+        "aten::divide",
+        "aten::true_divide",
+        "_operator::truediv",
+    )
+)
+def aten_div_complex(self: TFloat, other: TFloat) -> TFloat:
+    """div.Tensor(Tensor self, Tensor other) -> Tensor"""
+
+    # Complex division. PyTorch type promotion ensures both arguments are complex numbers
+    self_real = op.Slice(self, [0], [1], axes=[-1])
+    self_imag = op.Slice(self, [1], [2], axes=[-1])
+    other_real = op.Slice(other, [0], [1], axes=[-1])
+    other_imag = op.Slice(other, [1], [2], axes=[-1])
+
+    # Complex division
+    # (a + bi) / (c + di) = (ac + bd) / (c^2 + d^2) + (bc - ad) / (c^2 + d^2)i
+    # https://mathworld.wolfram.com/ComplexDivision.html
+    real = op.Div(
+        op.Add(op.Mul(self_real, other_real), op.Mul(self_imag, other_imag)),
+        op.Add(op.Mul(other_real, other_real), op.Mul(other_imag, other_imag)),
```

Comment:
nit: `op.Add(op.Mul(other_real, other_real), op.Mul(other_imag, other_imag)),` this part is repeated

Reply:
this is taking the power. did you see repeated code?

Reply:
repeated for real and imag?

Reply:
I see sorry. Let me update

Reply:
Done